bson/bsoncodec/bsoncodec.go

```diff
@@ -123,6 +123,16 @@ type DecodeContext struct {
 	// Ancestor is a bson.M, BSON embedded document values being decoded into an empty interface
 	// will be decoded into a bson.M.
 	Ancestor reflect.Type
```

Comment:
We should deprecate the `Ancestor` field with this change because it doesn't really do what it describes (e.g. it's effectively ignored when the `Decode` output parameter is a `map[string]interface{}`).

E.g.
```go
// Ancestor is a bson.M, BSON embedded document values being decoded into an empty interface
// will be decoded into a bson.M.
//
// Deprecated: Use DefaultDocumentType instead.
```

Reply:
Done!

Reply:
Now that the PR uses `DefaultDocumentM` and `DefaultDocumentD`, we should update the "Deprecated" message to say
> Deprecated: Use DefaultDocumentM or DefaultDocumentD instead.